src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -171,29 +181,38 @@ async Task InnerProcessMessages()
 
         Task ReceiveMessage()
         {
-            return TaskEx.Run(async state =>
+            // We pass a fake state to make sure we benefit from lamda delegate caching. See https://github.com/Particular/NServiceBus/issues/3884
+            var statePassedToAvoidLamdaDelegateCaching = this;
+
+            return Task.Factory.StartNew(async state =>
             {
-                var messagePump = (MessagePump)state;
+                var accessStateForPerfReasons = (MessagePump)state;
 
                 try
                 {
-                    await messagePump.receiveStrategy.ReceiveMessage().ConfigureAwait(false);
-                    messagePump.receiveCircuitBreaker.Success();
+                    await receiveStrategy.ReceiveMessage(messageProcessingCancellationTokenSource.Token).ConfigureAwait(false);
```

Comment:
won't this cause variable capturing again, removing the benefit to passing the state?